src/render/direct3d12/SDL_render_d3d12.c

```diff
@@ -32,6 +32,13 @@
 #include "../SDL_sysrender.h"
 #include "../SDL_d3dmath.h"
 
+/* Use C style c3d12 interfaces */
```

Comment:
I don't think this will work for GDK builds.

Reply:
Because AFAIK the headers are C++ only. I could be wrong though, @chalonverse may know better.

Reply:
I think this might work fine for Windows GDK, because it does use the normal Windows DirectX 12 headers. However, the Xbox version of the DirectX 12 headers (d3d12_x.h and d3d12_xs.h) do not implement `COBJMACROS`, and so removing the path where `D3D_CALL` does a C++ member function call will make the renderer no longer compile on either Xbox targets.

Reply:
@madebr I made a PR on your fork that should preserve all your changes while fixing it so that it still compiles on Xbox. I also had to add one additional `#define` to account for this change: https://github.com/libsdl-org/SDL/commit/17520c2e6eeac14839ecc510472353bf84a9c985 because DXGI_PRESENT_ALLOW_TEARING doesn't exist on Xbox targets